arch/x86/core/ia32/intstub.S

```diff
@@ -165,6 +165,7 @@ alreadyOnIntStack:
 
 #ifdef CONFIG_NESTED_INTERRUPTS
 	sti			/* re-enable interrupts */
+	nop
```

Comment:
This one is almost certainly safe, since the value in EDX is the ISR function pointer to invoke, and it can't reasonably be allowed to change an interrupt vector from within an ISR that has preempted the very ISR it's trying to change.  (No, I doubt we documented that limitation, but the whole idea is completely bonkers).  Given that this is on the interrupt entry path (i.e. we care about cycles more here than anywhere else), I guess I'd vote to remove it vs. being conservative.

Reply:
Coming back to amend, because I had the logic backwards: this **is** certainly safe, since EDX has already been loaded from memory.  The slot is capable of running with interrupts MASKED (i.e. before an expected interrupt has run that might otherwise be expected to change memory), so anything in registers must by definition reflect synchronized state.